### PR TITLE
feat(platform-packer-plugins): abstract our packer plugins util

### PR DESCRIPTION
**Pull request by BRKalow**
🎟️ [Asana Task]()

---

## Description

Abstracting out the utilities related to fetching content for packer plugins defined in a manifest file. I need access to these so that I can properly execute search indexing for packer in our shared workflows.

This was almost entirely lifted from https://github.com/hashicorp/dev-portal/tree/477e007fa880922170643001b1811aa49577fa83/src/components/_proxied-dot-io/packer/remote-plugin-docs, with a conversion to TS along the way.

## PR Checklist 🚀

- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Write a useful description (above) to give reviewers appropriate context.
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/packer-plugins/constants.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export const COMPONENT_TYPES = [
```
Comment (thiskevinwang) [inline packages/packer-plugins/constants.ts:1]:
```suggestion
/**
 * A list of Packer plugin component types. This maps to directory names within `./docs`:
 * - @see https://github.com/hashicorp/packer-plugin-scaffolding/tree/main/docs
 */
export const COMPONENT_TYPES = [
```

File: packages/packer-plugins/fetch-plugin-docs.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+// otherwise, return [err, null]
+// where err is an error message describing whether the
+// docs files were missing or invalid, with a path to resolution
+async function fetchDocsFiles({ repo, tag }: Options) {
+  // If there's a docs.zip asset, we'll prefer that
+  const docsZipUrl =
+    tag === 'latest'
+      ? `https://github.com/${repo}/releases/latest/download/docs.zip`
+      : `https://github.com/${repo}/releases/download/${tag}/docs.zip`
+  const docsZipResponse = await fetch(docsZipUrl, { method: 'GET' })
```
Comment (thiskevinwang) [inline packages/packer-plugins/fetch-plugin-docs.ts:26]: Q: is there a need for authenticated requests here? (w.r.t. rate limiting)

**Review by thiskevinwang — APPROVED** [review]

File: packages/packer-plugins/constants.ts (hunk 1 above)
Comment (BRKalow) [reply packages/packer-plugins/constants.ts:1]: Nice! Thanks.

File: packages/packer-plugins/fetch-plugin-docs.ts (hunk 2 above)
Comment (BRKalow) [reply packages/packer-plugins/fetch-plugin-docs.ts:26]: Good question, thus far I haven't run into any rate limiting / auth issues that I have noticed with these calls. The auth issues I think were stemming from when we were using the actual GH API to make un-authed API calls.

**Pull request merged**